Gems/Atom/RPI/Code/Source/RPI.Public/Pass/RasterPass.cpp

```diff
@@ -99,6 +99,116 @@ namespace AZ
+                    continue;
+                }
+
+                if (binding.m_scopeAttachmentUsage == RHI::ScopeAttachmentUsage::SubpassInput)
+                {
+                    AZ_Assert(subpassIndex > 0, "The first subpass can't have attachments used as SubpassInput");
+                    AZ_Assert(binding.m_unifiedScopeDesc.GetType() == RHI::AttachmentType::Image,
+                        "Only image attachments are allowed as SubpassInput.");
+                    AZ_Assert(m_subpassIndex > 0, "The first subpass can not depend on SubpassInput attachments");
+                    atLeastOneAttachmentWasSubpassInput = true;
+                    const auto aspectFlags = binding.m_unifiedScopeDesc.GetAsImage().m_imageViewDescriptor.m_aspectFlags;
+                    subpassLayoutBuilder.SubpassInputAttachment(binding.GetAttachment()->GetAttachmentId(), aspectFlags);
+                    continue;
+                }
+
+                // Skip bindings that aren't outputs or inputOutputs
+                if (binding.m_slotType != PassSlotType::Output && binding.m_slotType != PassSlotType::InputOutput)
+                {
+                    continue;
+                }
+
+                if (binding.m_scopeAttachmentUsage == RHI::ScopeAttachmentUsage::RenderTarget)
+                {
+                    RHI::Format format = binding.GetAttachment()->m_descriptor.m_image.m_format;
+                    subpassLayoutBuilder.RenderTargetAttachment(format, binding.GetAttachment()->GetAttachmentId());
+                }
+            }
+
+            if (subpassIndex > 0)
+            {
+                if (!atLeastOneAttachmentWasSubpassInput)
+                {
+                    AZ_Assert(false, "Starting from the second subpass, at least one attachment must be of SubpassInput usage.");
+                    return false;
+                }
+            }
+
+            return true;
+        }
+
+        bool RasterPass::SetRenderAttachmentLayout(
+            const AZStd::shared_ptr<RHI::RenderAttachmentLayout>& renderAttachmentLayout,
+            uint32_t subpassIndex)
+        {
+            if (!m_flags.m_mergeChildrenAsSubpasses)
+            {
+                AZ_Error("RasterPass", false, "%s can only be called on RasterPasses that are supposed to be merged as subpasses.", __FUNCTION__);
+                AZ_Assert(false, "%s can only be called on RasterPasses that are supposed to be merged as subpasses.", __FUNCTION__);
+                return false;
+            }
+        
+            if (subpassIndex != m_subpassIndex)
+            {
+                AZ_Error("RasterPass", false, "%s Was expecting subpassIndex=%u, instead got=%u.\n", __FUNCTION__, m_subpassIndex, subpassIndex);
+                AZ_Assert(false, "%s Was expecting subpassIndex=%u, instead got=%u.\n", __FUNCTION__, m_subpassIndex, subpassIndex);
```

Comment:
Not clear why the double error/assert, can we use just one?

Reply:
I left both because in Release builds the AZ_Asserts are removed. This error is clearly a programmer's error so in debug and profile it should assert. At least in Release there will be an explanation.